bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/tracer/BSpan.java

```diff
@@ -15,151 +15,116 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
 package io.ballerina.runtime.observability.tracer;
 
-import io.ballerina.runtime.observability.ObserverContext;
 import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMapExtractAdapter;
+import io.opentracing.propagation.TextMapInjectAdapter;
 
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
-import static io.ballerina.runtime.observability.tracer.TraceConstants.DEFAULT_OPERATION_NAME;
-import static io.ballerina.runtime.observability.tracer.TraceConstants.DEFAULT_SERVICE_NAME;
-import static io.ballerina.runtime.observability.tracer.TraceConstants.KEY_SPAN;
-
 /**
  * {@code BSpan} holds the trace of the current context.
  *
  * @since 0.964.1
  */
 public class BSpan {
-
-    private static final TraceManager manager = TraceManager.getInstance();
+    private final Tracer tracer;
+    private final Span span;
+
+    private BSpan(SpanContext parentSpanContext, String serviceName, String operationName, boolean isClientContext) {
+        tracer = TracersStore.getInstance().getTracer(serviceName);
```

Comment:
shall we remove the duplicity by moving below constructor and this ones logic to the static method, and make the constructors dumber.

Reply:
Fixed in 7795f8a